### PR TITLE
Final

### PR DESCRIPTION
**Pull request by sneeu-leeu**
#### In This Project we :
* Learned and applied HTML and CSS for media embedding
* Got to know flexBox
* Applied semantic HTML

by cloning a YouTube page

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+</head>
+
+<body>
+    <header>
+        <nav class="homeBar">
+            <div class="menu">
+                <i class="fas fa-bars"></i>
+            </div>
+            <div class="logo">
+                <i class="fab fa-youtube"></i>
```
Comment (expjazz) [inline index.html]: - [x] This should be an image:

image

- [x] The size could be smaller, so it matches the original:

| Yours | Original |
|:---:|:---:|
| image|image |

File: index.html [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+                    <div class="channel1">
+                        <img src="images/propic.jpg" alt="profile picture">
+                    </div>
+
+                    <div class="channel2">
+                        <h3>Tech News Today</h3>
+                        <div class="subs">
+                            <button id="subscribe">SUBSCRIBE</button>
+                            <small class="small">3.5M</small>
```
Comment (expjazz) [inline index.html]: - [x] The design could look more like the original. Try to pay more attention to the font-sizes, font-colors, and spacings.

| Yours | Original |
|:---:|:---:|
|image | image |

File: index.html [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+                    </div>
+                </div>
+
+                <hr style=": blue;" />
+
+                <div class="share">
+                    <div class="share-left">
+
+                        <i class="fas fa-plus"><small>add</small></i>
+                        <i class="title-icon fas fa-share">share</i>
```
Comment (expjazz) [inline index.html]: - [x] It needs to be a button:

image

File: stylesheet.css [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+
+.fas {
+    padding: 0 5px 0 5px;
+    font-size: 20px;
+    height: 25px;
+}
+
+.container {
+    display: flex;
+    display: -webkit-flex;
```
Comment (expjazz) [inline stylesheet.css]: Great job worrying about other browsers

File: stylesheet.css [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+}
+
+.channel1 {
+    float: left;
+}
+
+.channel2 {
+    float: left;
+}
+
```
Comment (expjazz) [inline stylesheet.css]: - [x] You could group those two selectors together since they have the same property

File: stylesheet.css [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+.share-right {
+    display: flex;
+    padding-left: 55%;
+}
+
+.share-left {
+    display: flex;
+}
```
Comment (expjazz) [inline stylesheet.css]: - [x] You can also group the share with the share-left

**Review by expjazz — CHANGES_REQUESTED** [review]
## Status: Changes Required :recycle: 

You did a very good job here. The page looks almost like the original :ok_hand: 

However, there are still changes to be made in order to make this project looks even better. For more detail, check the inline comments below. I will make a summary here:

- [x] There are a few repeated selectors. Group them together

- [ ] I saw that you have multiple branches. For the next project, work from one feature branch so you have more commits.

#### As stated above, the project looks really good.

### Please make the required changes and resubmit for a new code review once you are ready
Good Luck and Happy Coding :pray:

_P.S. If you have any questions about my review or need help, you can message me on slack (@ Expedito Andrade).:smile:_

File: stylesheet.css [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+    border-style: none;
+}
+.share-right {
+    display: flex;
+    padding-left: 55%;
+}
+
+.share-left {
+    display: flex;
+}
```
Comment (kevcap) [inline stylesheet.css]: - [x] Remove all the unnecessary repeated CSS code ([Good and Bad CSS Practices for Beginners](https://speckyboy.com/good-bad-css-practices/)) Tip: Create one only class like this: .d-flex{ display: flex } and use that same class in your index.html when you need it.

**Review by kevcap — CHANGES_REQUESTED** [review]
# [STATUS] CHANGES REQUIRED♻️

Hi team,  good job on making the previously required changes! 💪
You guys are almost there, but in order to accomplish as the actual page, you are required to do these changes listed below: ⬇

### I saw that you guys changed the image of the youtube logo and added a share button, however, you guys didn't implement the abruption of the css selectors, and that's why I can't approve this project. 

_Note_: That changes marked as [OPTIONAL] are not required to approve your project, but I strongly recommend that you guys go for that extra mile, in order to get more knowledge and have a better project to add at your portfolio.

[REQUIRED CHANGES]
- [x] Remove all the unnecessary repeated CSS code ([Good and Bad CSS Practices for Beginners](https://speckyboy.com/good-bad-css-practices/)). Explanation: You guys have a lot of repeated code, and it's not a good practice, remember to always code DRY (Don't Repeat Yourself).

[OPTIONAL CHANGES]
- [x] Your code is throwing some errors and warnings at the [HTML5 Validator](https://validator.w3.org/nu/#textarear), please fix them.
image

- [x] Make your whole layout more section more similar to the original one;
- [x] Put your `css` files in a css folder. (To make your repo more professional)
- [ ] Improve the layout to make it more similar to the original youtube page;
- [x] Improve the icons to make it more similar to the original youtube page;
- [ ] Add a favicon to make it more similar to the original youtube page;
- [ ] use `border-box` to add that bar on the top of the elements, instead of using `<hr>`
- [ ] Add the title of the video to the title of the page;
- [ ] Implement the comments section

**Please make the required changes and resubmit for a new code review once you are ready**💪


## As always, I wish you all the very best and see you both at the top🚀

👨🏻‍💻 **TSE: Kevin Alves**

  

📫 Have any question? Better call Kevin:

<a  href="mailto:kevinalves.js@gmail.com"> gmail</a> [Twitter Follow](https://twitter.com/intent/follow?screen_name=kevinnnmn) [](https://microverse-students.slack.com/team/UT86D3AP4)

  
<sub>*We're here to help you, to ensure that you're using the best practices and getting you ready for the job market. In real-world, you'll have a lot of changes requests for your real clients, it's good to get used to it.</sub>

File: css/style.css [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+    height: 100%;
+    flex-direction: row;
+    width: 100%;
+    padding-right: 40px;
+    border: 2px black;
+    font-size: 15px;
+}
+
+.views-name {
+    display: flex;
```
Comment (kevcap) [inline css/style.css]: - [x] Remove all the unnecessarily repeated CSS code. Keeping your code DRY is an excellent practice that will help you a lot and make your code more professional. (Check this link form better understanding [Good and Bad CSS Practices for Beginners](https://speckyboy.com/good-bad-css-practices/))

File: css/style.css [hunk 9]
```diff
@@ -0,0 +1,219 @@
+
+body {
+    margin: 10px;
+    padding: 0;
+    background-color: whitesmoke;
+    font-family: Roboto, Arial, sans-serif;
+}
+
+nav {
+    display: flex;
```
Comment (kevcap) [inline css/style.css]: - [ ] Remove all the unnecessarily repeated CSS code. Keeping your code DRY is an excellent practice that will help you a lot and make your code more professional. (Check this link form better understanding [Good and Bad CSS Practices for Beginners](https://speckyboy.com/good-bad-css-practices/))

File: css/style.css [hunk 10]
```diff
@@ -0,0 +1,219 @@
...
+    height:34px;
+}
+.homeBar {
+    width: 100%;
+    overflow: auto;
+
+}
+
+.logo {
+    display: flex;
```
Comment (kevcap) [inline css/style.css]: - [ ] Remove all the unnecessarily repeated CSS code. Keeping your code DRY is an excellent practice that will help you a lot and make your code more professional. (Check this link form better understanding [Good and Bad CSS Practices for Beginners](https://speckyboy.com/good-bad-css-practices/))

File: css/style.css [hunk 11]
```diff
@@ -0,0 +1,219 @@
...
+    align-items: center;
+    justify-content: space-between;
+}
+.logo img{
+    width: 100px;
+    height:50%;
+}
+
+.searchLink {
+    display: flex;
```
Comment (kevcap) [inline css/style.css]: - [ ] Remove all the unnecessarily repeated CSS code. Keeping your code DRY is an excellent practice that will help you a lot and make your code more professional. (Check this link form better understanding [Good and Bad CSS Practices for Beginners](https://speckyboy.com/good-bad-css-practices/))

File: css/style.css [hunk 12]
```diff
@@ -0,0 +1,219 @@
...
+    flex-wrap: wrap;
+}
+
+.searchBar {
+    width: 75%;
+    height: 28px;
+}
+
+.icons {
+    display: flex;
```
Comment (kevcap) [inline css/style.css]: - [ ] Remove all the unnecessarily repeated CSS code. Keeping your code DRY is an excellent practice that will help you a lot and make your code more professional. (Check this link form better understanding [Good and Bad CSS Practices for Beginners](https://speckyboy.com/good-bad-css-practices/))

File: css/style.css [hunk 13]
```diff
@@ -0,0 +1,219 @@
...
+
+.icons,
+.fas {
+    padding: 0 5px 0 5px;
+    font-size: 20px;
+    
+}
+
+.container {
+    display: flex;
```
Comment (kevcap) [inline css/style.css]: - [ ] Remove all the unnecessarily repeated CSS code. Keeping your code DRY is an excellent practice that will help you a lot and make your code more professional. (Check this link form better understanding [Good and Bad CSS Practices for Beginners](https://speckyboy.com/good-bad-css-practices/))

File: css/style.css [hunk 14]
```diff
@@ -0,0 +1,219 @@
...
+    width: 100%;
+}
+
+#subscribe {
+    background-color: red;
+    color: white;
+}
+
+.channel {
+    display: flex;
```
Comment (kevcap) [inline css/style.css]: - [ ] Remove all the unnecessarily repeated CSS code. Keeping your code DRY is an excellent practice that will help you a lot and make your code more professional. (Check this link form better understanding [Good and Bad CSS Practices for Beginners](https://speckyboy.com/good-bad-css-practices/))

File: css/style.css [hunk 15]
```diff
@@ -0,0 +1,219 @@
...
+
+.small {
+    border-color: black;
+    border-style: solid;
+    border: 2px;
+}
+
+.share,
+.share-right {
+    display: flex;
```
Comment (kevcap) [inline css/style.css]: - [ ] Remove all the unnecessarily repeated CSS code. Keeping your code DRY is an excellent practice that will help you a lot and make your code more professional. (Check this link form better understanding [Good and Bad CSS Practices for Beginners](https://speckyboy.com/good-bad-css-practices/))

**Review by kevcap — CHANGES_REQUESTED** [review]
# [STATUS] CHANGES REQUIRED♻️

Hi team,  good job on making the previously required changes! 💪
You guys are almost there, but in order to accomplish as the actual page, you are required to do these changes listed below: ⬇

_Note_: That changes marked as [OPTIONAL] are not required to approve your project, but I strongly recommend that you guys go for that extra mile, in order to get more knowledge and have a better project to add at your portfolio.

[REQUIRED CHANGES]
- [x] You guys still have a lot of repeated CSS code (e.g. `display: flex` repeats 9x) Remove all the unnecessary repeated CSS code ([Good and Bad CSS Practices for Beginners](https://speckyboy.com/good-bad-css-practices/)). 
Explanation: There's a lot of repeated code, and it's not a good practice, remember to always code DRY (Don't Repeat Yourself).
You guys have classes that do the same thing, yet they're separated. E.g.:
image
image
Those two classes add a display as flex, yet, they're separated. Kindly fix that boys! 💪

Tip: For example: Create one only class like this: 
```
.d-flex{ 
   display: flex
}
``` 
and use that same class in your HTML files when you need it. 
Apply the same logic (e.g. creating a `.w-100 { width: 100%}` and so on) of the example to **ALL** your repeated css rules listed below:

image


[OPTIONAL CHANGES]
- [x] Now your video size [1] and your description section links and share button [2] are both with layout broken, I encourage you to fix that
image

- [ ] Your page has a horizontal bar, you guys can fix that
image



- [x] Your code is still throwing some errors and warnings at the [HTML5 Validator](https://validator.w3.org/nu/#textarear), please fix them.
image

- [ ] Improve the layout to make it more similar to the original youtube page;
- [ ] Add a favicon to make it more similar to the original youtube page;
- [ ] use `border-box` to add that bar on the top of the elements, instead of using `<hr>`
- [x] Add the title of the video to the title of the page;
- [ ] Implement the comments section

**Please make the required changes and resubmit for a new code review once you are ready**💪

## As always, I wish you all the very best and see you both at the top🚀

👨🏻‍💻 **TSE: Kevin Alves**

  

📫 Have any question? Better call Kevin:

<a  href="mailto:kevinalves.js@gmail.com"> gmail</a> [Twitter Follow](https://twitter.com/intent/follow?screen_name=kevinnnmn) [](https://microverse-students.slack.com/team/UT86D3AP4)

  
<sub>*We're here to help you, to ensure that you're using the best practices and getting you ready for the job market. In real-world, you'll have a lot of changes requests for your real clients, it's good to get used to it.</sub>

**Review by SafaErden — APPROVED** [review]
### [STATUS] Approved 💪

<hr>
Hi TEAM, you did a great job and I can smell the great effort that you put for your very first collaborative project.😉

<hr>

- Since your project fulfills the requirements, I'm approving your project.

- But you have more ways to go and many things to learn. As a personal comment, I can say that you need to improve your UI abilities and I believe you will. 🌴 Try to make each project better than the previous one.

- Good luck with the next project.

<hr>

Don't forget to merge with the master branch

Happy coding 💻

**TSE:** Safa ERDEN

If you have any question, you can contact me on slack: @safaerden

**Pull request merged**